vector-im/element-android
# [Location sharing] - Show user live location pins in map view (PSF-1019)

## Type of change


## Content


Introducing `LocationSharingService` service interface on SDK side to group all location sharing related features.
On app side, display of all user pins currently sharing their live location in the map view opened when clicking on a message tile in the timeline.

## Motivation and context


Part of #6012 

## Screenshots / GIFs





<img src="https://user-images.githubusercontent.com/46314705/170264479-1165cd83-1d65-4609-be9b-8baff80104b1.png" width=300/><img src="https://user-images.githubusercontent.com/46314705/170264512-82e1f559-e0ef-4f5e-a0d9-c174b3222036.png" width=300/>

## Tests



It is suggested to test with a physical device since there may be some issues with pin display in MapView with emulator.
- Ensure the location sharing is enabled in Settings -> Preferences
- Ensure the live location sharing is enable in Settings -> Labs
- Go to a room
- Start sharing your live location
- Press the message when it is in the running state
- Check the pin is correctly displayed on the map and it is moving on location update

Special case to test:
- Since there is a database migration, try update the app from an old version of the app
- Ensure there is no crash at startup

## Tested devices

- OS version(s): Android 11

## Checklist
